Added joysticks math to the intake subsystem and IntakeRotaryJoysticks

zaraansari17 opened the pull request.

zaraansari17 committed:
Merge remote-tracking branch 'origin/main' into zansari

zaraansari17 committed:
Merge remote-tracking branch 'origin/main' into zansari

zaraansari17 committed:
Auto stash before merge of "zansari" and "origin/main"

Added methods to the intake subsystem

zaraansari17 committed:
Added joysticks math to the intake subsystem and IntakeRotaryJoysticks

jmullins16 committed:
Fixed up some missing changes and naming
- Some deletions were missing from the main branch (vision, etc.)
- Some naming wasn't quite consistent in the roller code--fixed
- Moved some global constants to be local
- Some vestigial imports and variables were deleted--in other subsystems

jmullins16 committed:
Fixed some naming inconsistencies and added a Phoenix v6 config for intake rotary

zaraansari17 committed:
Added to the periodic sim

zaraansari17 committed:
Merge branch 'zansari' of https://github.com/usfirst-frc-2135/robot2024 into zansari

zaraansari17 committed:
removed duplicates

zaraansari17 committed:
added the right joystick to moveRotaryWithJoystick

zaraansari17 committed:
Added mechanism for the rotary

zaraansari17 committed:
made name changes

zaraansari17 committed:
Added motion magic parameters and methods

jmullins16 reviewed (dismissed):
Looks good enough to merge. It has significant functionality that has started to work with no significant problems. We can make more changes/PRs going forward.

grace-mills reviewed (dismissed):
looks good

zaraansari17 committed:
progress trying to get correct angle of the intake rotary

jmullins16 commented on Comp2024/src/main/java/frc/robot/lib/math/Conversions.java:
```diff
@@ -25,7 +25,7 @@ public static double rotationsToOutputDegrees(double rotations, double gearRatio
    * @return Input Shaft Rotations
    */
-  public static double degreesToInputRotations(double degrees, double gearRatio)
+  public static double degreesToOutputRotations(double degrees, double gearRatio)
```
Generally, we shouldn't be changing the methods in the Conversions class. This method calculates the number of input rotations on a gearbox required to move a certain number of degrees on its output. If you need a different method, then write one to do the job. But the name of this one should be correct. (degreesToInputRotations)

jmullins16 commented on Comp2024/src/main/java/frc/robot/lib/util/CTREConfigs6.java:
```diff
@@ -88,7 +89,6 @@ public static CANcoderConfiguration exampleCancoderConfig( )
 
     // config.MagnetSensor.AbsoluteSensorRange = AbsoluteSensorRangeValue.Signed_PlusMinusHalf;
     // config.MagnetSensor.SensorDirection = SWConsts.steerCanCoderInvert;
-    // // config.MagnetSensor.MagnetOffset 
```
Since this is the example, we shouldn't delete this line.

jmullins16 commented on Comp2024/src/main/java/frc/robot/lib/util/CTREConfigs6.java:
```diff
@@ -154,6 +154,10 @@ public static TalonFXConfiguration intakeRotaryFXConfig( )
   public static CANcoderConfiguration intakeRotaryCancoderConfig( )
   {
     CANcoderConfiguration config = new CANcoderConfiguration( );
+
```
The easiest way to write a new config (CANcoder or TalonFX) is to copy the commented out lines from the example into your new config object. Then uncomment the ones needed. So there should be 3 lines below: one for AbsoluteSensorRange, SensorDirection, and MagnetOffset. You've got two.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -56,15 +56,15 @@ public class Intake extends SubsystemBase
   private final double              kLigament2dOffset   = 0.0; // Offset from mechanism root for wrist ligament
-  public static final double        kAngleMin           = -2.0;
+  public static final double        kAngleMin           = -90.0;
   public static final double        kAngleStow          = 0.0;
```
Delete/edit all angles down to: Min, retracted, handoff, deployed, max values. We will have the two end stops (min/max) and three functional positions (retracted-shooter, handoff to feeder, and deployed to intake).

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -150,7 +150,10 @@ public Intake( )
         PhoenixUtil6.getInstance( ).canCoderInitialize6(m_CANCoder, "Intake Rotary", CTREConfigs6.intakeRotaryCancoderConfig( ));
     if (Robot.isReal( ))
-      m_rotaryMotor.setPosition(Conversions.degreesToInputRotations(m_currentDegrees, kRotaryGearRatio));
+    {
```
Don't quite understand the need for this change, because reading the rotaryMotor when we start up is NOT guaranteed to be anywhere. Let this be done in the periodic method.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -239,7 +247,8 @@ public void initialize( )
   private double getTalonFXDegrees( )
   {
-    return Conversions.rotationsToOutputDegrees(m_rotaryPosition.refresh( ).getValue( ), kRotaryGearRatio);
+    m_rotaryPosition = m_rotaryMotor.getRotorPosition( ).refresh( );
```
I don't think this change is doing anything. The status signal we defined should make it update all of the time, and this worked on the wrist.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -290,6 +299,11 @@ private boolean isMoveValid(double degrees)
     return (degrees > kAngleMin) && (degrees < kAngleMax);
   }
 
+  public static double rotationsToDegrees(double rotations)
```
I think there's a Units library conversion that does this with writing our own.

jmullins16 reviewed (commented):
I know you're in the middle of hacking/debugging on this, but I wrote comments on what's there so far. Some of the changes weren't necessary.

jmullins16 committed:
Revert "progress trying to get correct angle of the intake rotary"

This reverts commit 2cc86a56116143df490fe188f987f471f3ee98dd.

jmullins16 committed:
Brought needed changes from previous commit for intake
- also performed some housekeeping on class member variables to remove unneeded and co-locate similar variables

jmullins16 reviewed (approved):
Ready for merge. Can make a new PR for Motion Magic.
